ethexe/observer/src/lib.rs

```diff
@@ -97,32 +104,59 @@ impl Stream for ObserverService {
     type Item = Result<ObserverEvent>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        if self.block_future.is_none() {
-            if let Poll::Ready(res) = self.headers_stream.poll_next_unpin(cx) {
-                if let Some(header) = res {
-                    self.block_future = Some(Box::pin(Self::get_block(
-                        header,
-                        self.provider.clone(),
-                        self.router,
-                    )));
-                } else {
-                    // TODO: test resubscribe works.
-                    log::warn!("Alloy headers stream ended, resubscribing");
-                    self.headers_stream = self.subscription.resubscribe().into_stream();
-                    cx.waker().wake_by_ref();
-                    return Poll::Pending;
-                }
+        if let Poll::Ready(res) = self.headers_stream.poll_next_unpin(cx) {
+            let Some(header) = res else {
+                // TODO (breathx): test resubscribe works.
+                log::warn!("Alloy headers stream ended, resubscribing");
+                self.headers_stream = self.subscription.resubscribe().into_stream();
+                cx.waker().wake_by_ref();
+                return Poll::Pending;
+            };
+
+            let data = SimpleBlockData {
+                hash: H256(header.hash.0),
+                header: BlockHeader {
+                    height: header.number as u32,
+                    timestamp: header.timestamp,
+                    parent_hash: H256(header.parent_hash.0),
+                },
+            };
+
+            log::trace!("Received a new block: {data:?}");
+
+            self.block_sync_queue.push_back(header);
+
+            return Poll::Ready(Some(Ok(ObserverEvent::Block(data))));
+        }
+
+        if self.sync_future.is_none() {
```

Comment:
shouldn't sync be prioritized?

Reply:
even if we don't send it to network? if we're unsynced we can't do anything anyway, but actually for most of cases it doesn't matter

Reply:
> if we're unsynced we can't do anything anyway

For example we can start to process off-chain transactions. Also BP could identify itself and starting to prepare for signature generation.

Reply:
we cannot start it if we're unsynced since we have another states rather than most of validators (if changed, but anyway)

Reply:
We can start in future when there will be some kind of off-chain consensus between validators. We must to do it in order to provide off-chain tx guaranties.